staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -108,6 +108,10 @@ type Reflector struct {
 	//
 	// See https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/3157-watch-list#design-details
 	UseWatchList bool
+	// watchListConsistencyDetector exists solely for testing purposes.
+	// The default value points to a function that is guarded by an environmental variable.
+	// Tests that wish to perform the consistency check bypass the existence of the env var.
+	watchListConsistencyDetector func(stopCh <-chan struct{}, identity string, lastSyncedResourceVersion string, listerWatcher Lister, store Store)
```

Comment:
how do you feel about it ?

Reply:
Given you always set it to checkWatchListConsistencyIfRequested in NewReflectorWithOptions, I don't see any value in it and it adds complexity.